v3/src/components/web-view/web-view-inspector.tsx

```diff
@@ -0,0 +1,63 @@
+import {useDisclosure} from "@chakra-ui/react"
+import {observer} from "mobx-react-lite"
+import React, {useRef, useState} from "react"
+import MediaToolIcon from "../../assets/icons/icon-media-tool.svg"
+import { useDocumentContext } from "../../hooks/use-document-context"
+import t from "../../utilities/translation/translate"
+import {InspectorButton, InspectorPanel} from "../inspector-panel"
+import {ITileInspectorPanelProps} from "../tiles/tile-base-props"
+import {WebViewUrlModal} from "./web-view-url-modal"
+import {isWebViewModel} from "./web-view-model"
+
+import "./web-view-inspector.scss"
+
+export const WebViewInspector = observer(function WebViewInspector({tile, show}: ITileInspectorPanelProps) {
+  const document = useDocumentContext()
+  const webViewModel = isWebViewModel(tile?.content) ? tile?.content : undefined
+  const panelRef = useRef<HTMLDivElement>()
+  const webViewModal = useDisclosure()
+  const [webViewModalIsOpen, setWebViewModalIsOpen] = useState(false)
+
+  const handleModalOpen = (open: boolean) => {
+    setWebViewModalIsOpen(open)
+  }
+
+  const handleSetWebViewUrlOpen = () => {
+    webViewModal.onOpen()
+    handleModalOpen(true)
+  }
+
+  const handleSetWebViewUrlClose = () => {
+    webViewModal.onClose()
+    handleModalOpen(false)
+  }
```

Comment:
I think you only need one of these mechanisms -- either `useDisclosure/onOpen/onClose` or the `isModalOpen/isModalClosed` state.